client/my-sites/customer-home/cards/features/blogging-prompt/prompts-navigation.jsx

```diff
@@ -24,7 +24,9 @@ const PromptsNavigation = ( { prompts } ) => {
 		return prompts !== undefined ? prompts[ promptIndex ] : null;
 	};
 
-	const getNewPostLink = () => addQueryArgs( editorUrl, { answer_prompt: getPrompt()?.id } );
+	// If no site ID set, go through site selector before rendering post editor
+	const getNewPostLink = () =>
+		addQueryArgs( siteId ? editorUrl : '/post', { answer_prompt: getPrompt()?.id } );
 
 	const goToPreviousStep = () => {
 		let nextIndex = promptIndex - 1;
```

Comment:
We should pull this out of `client/my-sites/customer-home/cards/features/` if it's going to be used in both My Sites and Reader. Maybe somewhere in `client/components`? What do you think?

Reply:
This is a good idea too, I'll create a task to handle this in a follow up PR. Thanks.

Reply:
https://github.com/Automattic/wp-calypso/issues/73158